src/main/java/de/rytrox/varo/Varo.java

```diff
@@ -150,4 +142,47 @@ private DatabaseConfig getDatabaseConfig() {
 
         return config;
     }
+
+    /**
+     * Creates instances of all services and groups them in GameStates
+     */
+    private void registerServices() {
+
+        DiscordListener discordListener = new DiscordListener(this);
+        this.moderatorManager = new ModeratorManager(this);
+        ModeratorTeleporter moderatorTeleporter = new ModeratorTeleporter(this, moderatorManager);
+        PlayerResurrectionListener playerResurrectionListener = new PlayerResurrectionListener();
+        PlayerSkullDropService playerSkullDropService = new PlayerSkullDropService();
+        this.scoreBoardManager = new ScoreBoardManager(this);
+        GameTimeService gameTimeService = new GameTimeService(this);
+        this.teamManager = new TeamManager(this);
+        TeamMemberGhostService teamMemberGhostService = new TeamMemberGhostService(this);
+
+        TeamMemberRepository teamMemberRepository = new TeamMemberRepository(getDB());
+
+        TeamInventoryManager teamInventoryManager = new TeamInventoryManager(this, teamMemberRepository);
+        SpawnPointService spawnPointService = new SpawnPointService(this, teamMemberRepository);
+        this.worldBorderHandler = new WorldBorderHandler(this);
+
+
+        GameState gameStateSetup = new GameState("Setup", teamManager, spawnPointService);
+        GameState gameStatePreGame = new GameState("Pre-Game", discordListener, moderatorManager, moderatorTeleporter, scoreBoardManager, spawnPointService);
+        GameState gameStateMain = new GameState("Main",
+                discordListener, moderatorManager, moderatorTeleporter,
+                playerResurrectionListener, playerSkullDropService, scoreBoardManager,
+                gameTimeService, teamMemberGhostService, teamInventoryManager);
+        GameState gameStateFinal = new GameState("Final",
+                discordListener, moderatorManager, moderatorTeleporter,
+                playerResurrectionListener, playerSkullDropService, scoreBoardManager,
+                gameTimeService, teamMemberGhostService, teamInventoryManager);
+        GameState gameStatePost = new GameState("Post",
+                moderatorManager, moderatorTeleporter, scoreBoardManager);
+
+        GameStateHandler.getInstance(this).registerGameStates(
+                gameStateSetup,
+                gameStatePreGame,
+                gameStateMain,
+                gameStateFinal,
+                gameStatePost);
+    }
```

Comment:
Was ich hier zu kritisieren hab, ist, dass es viel zu unübersichtlich bei der Initialisierung ist. Können wir das nicht so machen, dass wir es zulassen, dass wir CustomGameStates als Vererbungsstrategie zulassen?

Reply:
ich dachte mir schon, dass du Vererbungen haben möchtest, allerdings finde ich das absolut unsinnig, weil jeder GameState exakt das gleiche macht. Außerdem habe ich hier den Vorteil, dass ich jeden Service nur einmal instanziieren muss und nicht in jedem GameState neu.

Reply:
Das muss ich im GameState bei Vererbung auch, das ist kein Punkt. Mag sein, dass der Handler dasselbe macht, aber es sind unterschiedliche Listener die registriert werden müssen, daher gefällt mir hier die Idee der Vererbung für die Übersicht besser. Registrieren muss ich meinen GameState nur einmalig im Handler, daher verstehe ich hier deinen Punkt mit Mehrfachinitialisierung nicht

Reply:
Falls es noch nicht angekommen sein sollte, ich binde hier nicht deine GameState-Strategie ein :d

Reply:
Ich weiß, trotzdem sollte man das offen lassen. Verseh die Interne Logik mit final, die sich nicht ändert und ich pass da auch mal einen Vorschlag an. Der Handler ist ja an sich ne schöne Sache und von GameState kann man ja theoretisch erben und das ohne Probleme.

Reply:
Mit der "Mehrfachinitialisierung" beziehe ich mich nicht auf die gamestates, sondern auf die Listener. Die müsste ich ja dann in jedem GameState neu erstellen. Wobei man das auch umgehen könnte, wenn ich die funktionalität noch erweitern würde. Aber das würde vermutlich wieder der Übersicht schaden...

Reply:
Nicht unbedingt, ich kann doch einfach Dependency-Injection machen, oder ist da irgendwas, was mich dran hindert?

Reply:
du nervst mich langsam...
ich lass das jetzt liegen und überleg mir noch was. Deine Vererbungen kriegst du aber nicht. das kannste vergessen